src/zuko/entity/reader.cljc

```diff
@@ -96,15 +96,15 @@
      (persistent!
       (reduce (fn [m [k v]]
                 (assoc! m k (if-let [[km vm] (find m k)]
-                              (if (set? vm) (conj vm v) #{vm v})
+                              (if (set? vm) (conj vm v) (hash-set vm v))
                               v)))
               (transient {}) kvs))
      :cljs
      (persistent!
       (reduce (fn [m [k v]]
                 (assoc! m k (let [vm (get m k ::null)]
                               (if-not (= ::null vm)
-                                (if (set? vm) (conj vm v) #{vm v})
+                                (if (set? vm) (conj vm v) (hash-set vm v))
```

Comment:
Technically, this patch as is not needed as ClojureScript does not have this problem, but I included it for consistency.